src/runtime/embassy.rs

```diff
@@ -7,118 +7,89 @@ pub mod runtime {
     pub const STANDARD_MESSAGE_CHANNEL_SIZE: usize = DEFAULT_CHANNEL_SIZE;
-    pub type StandardMessageMutex = DefaultChannelMutex;
     pub type StandardMessageHandle =
-        EmbassyHandle<StandardMessageMutex, StandardMessage, STANDARD_MESSAGE_CHANNEL_SIZE>;
-    pub type StandardMessagePool =
-        HandlePool<StandardMessageMutex, StandardMessage, STANDARD_MESSAGE_CHANNEL_SIZE>;
+        EmbassyHandle<StandardPayload, StandardMessageChannelMutex, STANDARD_MESSAGE_CHANNEL_SIZE>;
+    pub type StandardMessagePool = ChannelPool<StandardMessageHandle>;
 
-    pub struct HandlePool<Mutex: RawMutex + 'static, Msg: ActorMessage, const Q: usize> {
-        pool: RefCell<Vec<(EmbassyHandle<Mutex, Msg, Q>, bool)>>,
+    /// An entry in our pool: a single channel plus "in_use" flag
+    struct PooledHandle<H> {
+        handle: H,
+        in_use: bool,
     }
 
-    impl<Mutex, Msg, const Q: usize> HandlePool<Mutex, Msg, Q>
+    pub struct ChannelPool<H>
     where
-        Mutex: RawMutex + 'static,
-        Msg: ActorMessage,
+        H: Clone + Send,
     {
+        pool: RefCell<Vec<PooledHandle<H>>>,
+    }
+
+    impl<M: 'static, Mutex: RawMutex + Sync + 'static, const Q: usize>
+        ChannelPool<EmbassyHandle<M, Mutex, Q>>
+    where
+        M: Send,
+    {
+        /// Creates a pool of `capacity` distinct channels, each leaked to `'static`.
         pub fn new(capacity: usize) -> Self {
-            let mut pool = Vec::with_capacity(capacity);
-            for _ in 0..capacity {
-                let channel: &'static _ = Box::leak(Box::new(Channel::new()));
-                let handle = EmbassyHandle::new(channel);
-                pool.push((handle, false));
+            let mut vec = Vec::with_capacity(capacity);
+
+            for i in 0..capacity {
+                let channel: &'static Channel<Mutex, Message<M>, Q> =
+                    Box::leak(Box::new(Channel::new()));
+                let handle = Handle::new(i, channel);
+
+                vec.push(PooledHandle {
+                    handle,
+                    in_use: false,
+                });
             }
-            HandlePool {
-                pool: RefCell::new(pool),
+
+            Self {
+                pool: RefCell::new(vec),
             }
         }
 
-        pub fn acquire(&self) -> Option<EmbassyHandle<Mutex, Msg, Q>> {
-            let mut pool = self.pool.borrow_mut();
-            for (handle, in_use) in pool.iter_mut() {
-                if !*in_use {
-                    *in_use = true;
-                    return Some(handle.clone());
+        /// Acquire an unused channel from the pool
+        pub fn acquire(&self) -> Option<EmbassyHandle<M, Mutex, Q>> {
+            let mut guard = self.pool.borrow_mut();
+            for entry in guard.iter_mut() {
+                if !entry.in_use {
+                    entry.in_use = true;
+                    return Some(entry.handle.clone());
```

Comment:
This is effectively `iter_mut().find()` I believe. Or something very close to it.